sdks/python/apache_beam/runners/interactive/interactive_beam.py

```diff
@@ -220,6 +220,24 @@ def display_timezone(self, value):
     """
     self._display_timezone = value
 
+  @property
+  def specified_cache_dir(self):
+    """The cache directory specified by the user.
+
+    Defaults to None.
+    """
+    return self._specified_cache_dir
+
+  @specified_cache_dir.setter
```

Comment:
This setter doesn't seem to take effect immediately for existing cache managers.

There can be 2 behaviors:
1. It overrides the cache dir for all existing ie.current_env()._cache_managers, evict all existing cache files (locally or on GCS, the GCS file deletion can be done in later PRs but add a TODO here) and log warnings.
2. Notify the user that this only takes effect for new pipelines: restarting the kernel or re-execute pipeline creation codes.

Reply:
I would say #2. We want to be careful with the user files in case they don't want them deletec.

Reply:
I've added new logger warnings that are displayed if interactive_beam.options.cache_root has been set. These warnings can be found in background_caching_job.py where the StreamingCache is initialized, and in interactive_environment.py where the FileBasedCacheManager is initialized.